server/src/main/java/org/elasticsearch/index/codec/bloomfilter/ES84BloomFilterPostingsFormat.java

```diff
@@ -0,0 +1,479 @@
+                    } else {
+                        return false;
+                    }
+                }
+
+                @Override
+                public void seekExact(BytesRef term, TermState state) throws IOException {
+                    getDelegate().seekExact(term, state);
+                }
+
+                @Override
+                public TermState termState() throws IOException {
+                    // TODO: return TermState that includes BloomFilter and fix _disk_usage API
+                    return getDelegate().termState();
+                }
+            };
+        }
+    }
+
+    private abstract static class LazyFilterTermsEnum extends BaseTermsEnum {
+        abstract TermsEnum getDelegate() throws IOException;
+
+        @Override
+        public SeekStatus seekCeil(BytesRef text) throws IOException {
+            return getDelegate().seekCeil(text);
+        }
+
+        @Override
+        public void seekExact(long ord) throws IOException {
+            getDelegate().seekExact(ord);
+        }
+
+        @Override
+        public BytesRef term() throws IOException {
+            return getDelegate().term();
+        }
+
+        @Override
+        public long ord() throws IOException {
+            return getDelegate().ord();
+        }
+
+        @Override
+        public int docFreq() throws IOException {
+            return getDelegate().docFreq();
+        }
+
+        @Override
+        public long totalTermFreq() throws IOException {
+            return getDelegate().totalTermFreq();
+        }
+
+        @Override
+        public PostingsEnum postings(PostingsEnum reuse, int flags) throws IOException {
+            return getDelegate().postings(reuse, flags);
+        }
+
+        @Override
+        public ImpactsEnum impacts(int flags) throws IOException {
+            return getDelegate().impacts(flags);
+        }
+
+        @Override
+        public BytesRef next() throws IOException {
+            return getDelegate().next();
+        }
+
+        @Override
+        public AttributeSource attributes() {
+            try {
+                return getDelegate().attributes();
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+    }
+
+    static int bloomFilterSize(int maxDocs) {
+        // 10% saturation (i.e., 10 bits for each term)
+        final long numBits = maxDocs * 10L;
+        if (numBits > Integer.MAX_VALUE) {
+            return Integer.MAX_VALUE;
+        } else {
+            return (int) numBits;
+        }
+    }
+
+    static int numBytesForBloomFilter(int bloomFilterSize) {
+        return Math.toIntExact((bloomFilterSize + 7L) / 8L);
+    }
+
+    static int hashTerm(BytesRef br) {
+        final int hash = murmurhash3_x86_32(br.bytes, br.offset, br.length, 0x9747b28c);
+        return hash < 0 ? -hash : hash;
+    }
+
+    /**
+     * Forked from Lucene's StringHelper#murmurhash3_x86_32
```

Comment:
```suggestion
     * Forked from Lucene's StringHelper#murmurhash3_x86_32 so that changes to the Lucene implementation do not break the compatibility of this format.
```